services/user.service.js

```diff
@@ -0,0 +1,72 @@
+const User = require("../models/user.model");
+const sessionHelper = require("../helpers/sessionHelper");
+
+async function findUserByWalletAddress(walletAddress) {
+	const user = await User.find({ walletAddress: walletAddress });
+	if (!user) {
+		throw new Error("User not found");
+	}
+	return user;
+}
+
+async function findUserById(userId) {
+	const user = await User.findById(userId);
+	if (!user) {
+		throw new Error("User not found");
+	}
+	return user;
+}
+
+async function findAllUsers() {
+	const users = await User.find();
+	if (!users) {
+		throw new Error("Users not found");
+	}
+	return users;
+}
+
+async function createUser(walletAddress) {
+	const newUser = new User({
+		username: walletAddress,
+		walletAddress: walletAddress,
+	});
+	const savedUser = await newUser.save();
+	return savedUser;
+}
+
+async function findAndRegisterUser(req, walletAddress) {
+	let user = await findUserByWalletAddress(walletAddress);
+	sessionHelper.setSessionUser(req, user[0]);
+	return user[0];
+}
+
+async function createAndRegisterUser(req, walletAddress) {
+	let newUser = await createUser(walletAddress);
+	sessionHelper.setSessionUser(req, newUser);
+	return newUser;
+}
+
+async function registerOrCreateUser(req, walletAddress) {
+	let user = await findUserByWalletAddress(walletAddress);
+	if (user.length === 0) {
+		return await createAndRegisterUser(req, walletAddress);
+	} else {
+		return await findAndRegisterUser(req, walletAddress);
```

Comment:
This should have the name findAndLogin and the other should be createAndRegister, the reason you should not put User at the end of these is because you are already inside user.service. So you should always use these functions as UserService.createAndRegister() etc...